sda-commons-shared-otel-agent/build.gradle

```diff
@@ -0,0 +1,57 @@
+plugins{
+  id "com.github.johnrengelman.shadow" version "7.1.2"
+}
+ext {
+  otelAgentVersion = '1.13.1'
+  otelAgentAlphaVersion = '1.13.1-alpha'
+
+  autoservice= dependencies.create(group: 'com.google.auto.service', name: 'auto-service', version: '1.0')
+}
+
+configurations {
+  agents
+  // Interfaces that are used to compile the SPI are also needed for testing
+  testImplementation.extendsFrom compileOnly
```

Comment:
strange...

Reply:
True. At first most of the dependencies were only needed at compile time, this one was only added to supports tests as well, but that can be avoided. I will simplify it.